skimage/exposure/exposure.py

```diff
@@ -346,6 +346,14 @@ def rescale_intensity(image, in_range='image', out_range='dtype'):
     imin, imax = intensity_range(image, in_range)
     omin, omax = intensity_range(image, out_range, clip_negative=(imin >= 0))
 
+    if np.isnan(imin + imax + omin + omax):
```

Comment:
This seems like a hack. But otherwise ok

Reply:
Is it a bad hack? :sweat_smile: I find it clear enough with the warning message directly below. It's faster than other options but seems insignificant when timing the whole function.

Reply:
Just a hack. Can't comment if good or bad. But maybe requires a comment to those new to scientific computing.

Reply:
I personally prefer `np.isnan([imin, imax, omin, omax]).any()`, but it's fine :-)

Reply:
Thanks for the comment up there. I'm OK with this now.